tests/unittests/core/worker/test_trials_history.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Collection of tests for :mod:`orion.core.worker.trials_history`."""
+
+from orion.core.worker.trials_history import TrialsHistory
+
+
+class DummyTrial(object):
+    """Mocking class for the Trial"""
+
+    def __init__(self, trial_id, parents):
+        """Init _id and parents only"""
+        self.id = trial_id
+        self.parents = parents
+
+
+def test_added_children_without_ancestors():
+    """Verify that children are added to history"""
+    trials_history = TrialsHistory()
+    trials_history.update([DummyTrial(i, []) for i in range(3)])
+    assert trials_history.children == [0, 1, 2]
+    trials_history.update([DummyTrial(i, []) for i in range(3, 6)])
+    assert trials_history.children == [0, 1, 2, 3, 4, 5]
+
+
+def test_added_children_with_ancestors():
+    """Verify that children with ancestors are added to history"""
+    trials_history = TrialsHistory()
+    trials = [DummyTrial(i, []) for i in range(3)]
+    trials_history.update(trials)
+    assert trials_history.children == [0, 1, 2]
+
+    trials = [DummyTrial(i, [trials[i % 3].id]) for i in range(3, 6)]
+    trials_history.update(trials)
+    assert len(set(trials_history.children) & set([3, 4, 5])) == 3
```

Comment:
shouldn't this just be `set(trials_history.children) == set([3, 4, 5])`

Reply:
Not exactly. I wanted to avoid testing that the duplicates are properly removed, which is the role of `test_discarded_children`. That's why I only test that (3, 4, 5) are present in `trials_history.children`, without testing if other trials are still there or discarded properly.

Reply:
Ahh, makes sense!